ProfileManager.js

```diff
@@ -84,6 +92,118 @@ export default class ProfileManager {
     await this._sessionChanged({newData: session.data});
   }
 
+  // TODO: add docs
+  // FIXME: rename?
+  // create a capabilitySet EDV and create a document in that EDV and create
+  // delegations related to that document
+  async createCapabilitySetEdv({profileId, referenceId, zcaps}) {
+    const {profileAgent} = await this._profileService.getAgentByProfile({
+      account: this.accountId,
+      profile: profileId
+    });
+    const {invocationSigner, kmsClient} = await this.getProfileSigner({
+      profileId,
+      profileAgentId: profileAgent.id
+    });
+    const edv = await edvs.create({
+      invocationSigner,
+      kmsClient,
+      kmsModule: this.kmsModule,
+      profileId,
+      referenceId,
+      edvBaseUrl: this.edvBaseUrl,
+    });
+
+    // create the capabilitySet document for the profile agent
+    const capabilitySetDocument = await edv.insert({
+      doc: {
+        content: {
+          // FIXME:
+          // WHAT OTHER PROPERTIES ARE NEEDED
+          // WE MAY NEED DETAILS THAT ASSIST IN IDENTIFYING CAPABILITYSET DOCS
+          // FOR THE PURPOSE OF DELETING THEM FOR INSTANCE.
+          zcaps,
+        }
+      },
+      invocationSigner,
+    });
+
+    const delegateEdvDocumentRequest = {
+      referenceId: `${referenceId}-edv-document`,
+      // FIXME: does allowedAction need to be here or would it be appropriate
+      // to remove
+      allowedAction: ['read', 'write'],
+      controller: profileAgent.id,
+      invocationTarget: {
+        id: `${edv.id}/documents/${capabilitySetDocument.id}`,
+        type: 'urn:edv:documents'
```

Comment:
zcap is for a *specific* document only (invocation target is a document, not a collection of them).

```suggestion
        type: 'urn:edv:document'
```